src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -72,11 +73,19 @@ def create_resource(self, name, **kwargs):
             self.test_class_instance.kwargs[self.key] = self.dev_setting_name
             return {self.parameter_name: self.dev_setting_name,
                     self.parameter_name_for_location: self.dev_setting_location}
-
+        test_class_path = sys.modules[self.test_class_instance.__module__].__file__.split(os.sep)
+        # get index of the module name for main repo
+        if 'command_modules' in test_class_path:
+            index_of_module = test_class_path.index('command_modules') + 1
+        # get index of the extension name for extension repo
+        else:
```

Comment:
Maybe use the following code to make the code more robust (in case there are other scenarios besides `command_moduels` and `src`.

```python
else if `src` in test_class_path:
   index_of_module = test_class_path.index('src') + 1
else:
   xxx
```